File: linkedlist.py
#!/usr/bin/env python3.4

# This is a singly linked,linked list.

class node: # This is the simplest element in the linked list

	def __init__(self, key):
		"""Init the values of the node"""
		self.key = key
		self.next = None

class stack:

	def __init__(self):
		"""Init the head of the list"""
		self.head = None

	def push(self, key):
		"""function to insert element into the linked list"""
		
		newnode = node(key)

		if self.head is None:
			self.head = newnode

		else:
			# Since this is a stack, the insertion is only at the front (and yeah thats what she said :( )
			newnode.next = self.head
			self.head = newnode

	def pop(self):
		"""Function to delete elements from the linked list"""

		if self.head is None:
			print("There is nothing to delete from the list. Consider adding some elements.")

		else:
			print("Popped element(stack): ", self.head.key)
			self.head = self.head.next

	def display(self):
		"""Display all the keys in the linked list"""

		current_node = self.head # Cause we start from there

		print("\nStack: ")
		while(current_node is not None):
			if(current_node.next is None):
				print(current_node.key)
				current_node = None

			else:
				print(current_node.key, end = " -> ")
				current_node = current_node.next

class queue:

	def __init__(self):
		"""Init the head of the list"""
		self.head = None

	def enqueue(self, key):
		"""function to insert element into the linked list"""
		
		newnode = node(key)

		if self.head is None:
			self.head = newnode

		else:
			# Since this is a stack, the insertion is at the end of the linked list
			# So we have to first find the end of the linked list

			current_node = self.head

			while(current_node.next is not None):
				current_node = current_node.next

			current_node.next = newnode

	def dequeue(self):
		"""Function to delete elements from the linked list"""

		if self.head is None:
			print("There is nothing to delete from the list. Consider adding some elements.")

		else:
			print("Removed element(queue): ", self.head.key)
			self.head = self.head.next

	def display(self):
		"""Display all the keys in the linked list"""

		current_node = self.head # Cause we start from there

		print("\nQueue: ")
		while(current_node is not None):
			if(current_node.next is None):
				print(current_node.key)
				current_node = None

			else:
				print(current_node.key, end = " -> ")
				current_node = current_node.next

stk = stack()
q = queue()

again = True

while(again):
    a = int(input("\nOptions \n1. Push\n2. Pop\n3. Display\n4. Exit\n  "))

    if a == 1:
        key = int(input("Key: "))
        stk.push(key)
        q.enqueue(key)

    elif a == 2:
        stk.pop()
        q.dequeue()

    elif a == 3:
        stk.display()
        q.display()

    elif a == 4:
        again = False
    
    else:
        print("\nWrong option selected. Try again.")

File: linkedlist.cpp
#include <iostream>
using namespace std;

class node{
	public:
	
	int key;
	node * next;

	void insert_key(int data) {
	 	key = data;
		next = NULL;
	}
};

class stack{
	node * head = new node();

public:

	stack() {
		head = NULL; //Since this doesnt point to anything during init
	}

	void push(int data) {
		
		//Make a newnode and init
		node * newnode = new node();
		newnode->insert_key(data);

		if(head == NULL) {
			//Then the newnode will become the head
			head = newnode;
		}

		else {
			newnode->next = head;
			head = newnode;
		}
	}

	void pop() {
		if (head == NULL) {
			cout<<"\nThere is nothing in the stack to delete.";
		}

		else {
			cout<<"\nPopped element(stack): "<<head->key;
			head = head->next;
		}
	}

	void display() {
		cout<<"\nStack: ";
		node * current_node = new node ();
		current_node = head;

		while(current_node != NULL) {
			if (current_node->next == NULL) {
				cout<<current_node->key;
			}

			else {
				cout<<current_node->key<<" -> ";
			}

			current_node = current_node->next;
		}
	}
};

class queue{
	node * head = new node();

public:

	queue() {
		head = NULL; //Since this doesnt point to anything during init
	}

	void enqueue(int data) {
		
		//Make a newnode and init
		node * newnode = new node();
		newnode->insert_key(data);

		if(head == NULL) {
			//Then the newnode will become the head
			head = newnode;
		}

		else {

			//Here we have to find the end of the linked list to insert the node
			node *current_node = new node();
			current_node = head;
			
			while(current_node->next != NULL) {
				current_node = current_node->next;
			}

			current_node->next = newnode;
		}
	}

	void dequeue() {
		if (head == NULL) {
			cout<<"\nThere is nothing in the queue to delete.";
		}

		else {
			cout<<"\nRemoved element(queue): "<<head->key;
			head = head->next;
		}
	}

	void display() {
		cout<<"\nQueue: ";
		node * current_node = new node ();
		current_node = head;

		while(current_node != NULL) {
			if (current_node->next == NULL) {
				cout<<current_node->key;
			}

			else {
				cout<<current_node->key<<" -> ";
			}

			current_node = current_node->next;
		}
	}
};

int main() {
	stack stk;
	queue q;
	bool again = true;

	while(again) {
		int c;
		cout<<"\n\nOptions:\n1. Insert\n2. Delete\n3. Display\n4. Exit\n";
		cin>>c;

		switch(c) {
			case 1:	int data;
					cout<<"\nData: ";
					cin>>data;
					
					stk.push(data);
					q.enqueue(data);

					break;

			case 2: stk.pop();
					q.dequeue();
					break;

			case 3: stk.display();
					q.display();
					break;

			case 4: again = false;
					break;

			default: cout<<"\nWrong option. Try again";
		}
	}

	return 0;
}